.github/workflows/tests.yml

```diff
@@ -623,6 +681,21 @@ jobs:
         sudo apt install libgmp-dev patchelf
         python -m pip install --upgrade pip wheel
         make install_devel
+    - name: Connect to Memcached server
+      if: ${{ needs.skip_check_apps.outputs.should_skip != 'true' }}
+      shell: bash
+      env:
+        SSH_KEY: ${{ secrets.MEMCACHED_SSH_KEY }}
+        SERVER: ${{ secrets.MEMCACHED_SERVER }}
+      run: |
+        eval $(ssh-agent -s)
+        echo "$SSH_KEY" | tr -d '\r' | ssh-add - &>/dev/null
+        mkdir -p ~/.ssh
+        chmod 700 ~/.ssh
+        ssh-keyscan $SERVER >> ~/.ssh/known_hosts
+        chmod 644 ~/.ssh/known_hosts
+        ssh -L 11211:localhost:11211 -N -T -f -o ExitOnForwardFailure=yes -o ServerAliveInterval=10 -o ServerAliveCountMax=3 ci@$SERVER &>/dev/null
+        echo "stats" | nc -q 1 localhost 11211
```

Comment:
Is there a way to get rid of these 5 redundant sections in GitHub actions?

Reply:
In the past, it was not possible to reuse steps, but it seems that has changed in the meantime: https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/. I will have a look at it.

Reply:
I have created #850 to reduce duplication in our GitHub actions in general.